build/room_ensurer/room_ensurer.py

```diff
@@ -205,9 +210,42 @@ def _ensure_room_for_network(self, network: Networks, alias_fragment: str) -> No
                     for server_name, room_info in room_infos.items()
                 },
             )
+            return
+
+        # There are either mis
+        own_server_room_info = room_infos.get(self._own_server_name)
+        own_server_room_alias = f"#{room_alias_prefix}:{self._own_server_name}"
+        first_server_room_alias = f"#{room_alias_prefix}:{self._first_server_name}"
+
+        if not own_server_room_info:
+            log.warning(
+                "Room missing on own server, adding alias",
+                server_name=self._own_server_name,
+                room_id=ensure_room_info.room_id,
+                new_room_alias=own_server_room_alias,
+            )
+            self._join_and_alias_room(first_server_room_alias, own_server_room_alias)
+            log.info("Room alias set", alias=own_server_room_alias)
+
+        elif own_server_room_info.room_id != ensure_room_info.room_id:
+            log.warning(
+                "Conflicting local room, reassigning alias",
+                server_name=self._own_server_name,
+                expected_room_id=ensure_room_info.room_id,
+                current_room_id=own_server_room_info.room_id,
+            )
+            self._own_api.remove_room_alias(own_server_room_alias)
+            self._join_and_alias_room(first_server_room_alias, own_server_room_alias)
+            log.info(
+                "Room alias updated",
+                alias=own_server_room_alias,
+                room_id=ensure_room_info.room_id,
+            )
+        else:
+            log.warning("Mismatching rooms on other servers. Doing nothing.")
 
         # Ensure that all admins have admin power levels
-        self._ensure_admin_power_levels(room_infos[self._own_server_name])
+        self._ensure_admin_power_levels(own_server_room_info)
```

Comment:
I'm not sure in which cases this has to be done. Is it ok to skip in all cases where this function returns early?

Reply:
Indeed I found a case when it should not be skipped. Thanks for asking. Ensuring power_levels should always happen if the user is part of the expected room.

The function returned early if the state is ok but it still should ensure power levels. I removed the return